src/util/strencodings.cpp

```diff
@@ -526,3 +527,45 @@ std::string HexStr(const Span<const uint8_t> s)
     assert(it == rv.end());
     return rv;
 }
+
+std::optional<uint64_t> ParseByteUnits(const std::string& str, uint64_t default_multiplier)
+{
+    auto multiplier = default_multiplier ? default_multiplier : 1;
+    char unit = str.empty() ? 0 : str.back();
+    switch (unit) {
+    case 'k':
+        multiplier = 1000;
+        break;
+    case 'K':
+        multiplier = 1024;
+        break;
+    case 'm':
+        multiplier = 1000 * 1000;
+        break;
+    case 'M':
+        multiplier = 1ULL << 20;
+        break;
+    case 'g':
+        multiplier = 1000ULL * 1000 * 1000;
+        break;
+    case 'G':
+        multiplier = 1ULL << 30;
+        break;
+    case 't':
+        multiplier = 1000ULL * 1000 * 1000 * 1000;
+        break;
+    case 'T':
+        multiplier = 1ULL << 40;
+        break;
+    default:
+        unit = 0;
+        break;
+    }
+
+    uint64_t num;
+    if (!ParseUInt64(unit ? str.substr(0, str.size() - 1) : str, &num) ||
```

Comment:
With this breaking change (no longer ignoring whitespace), which I think is fine, it might be best to also reject sign characters (+-) and use `ToIntegral<uint64_t>`.

Reply:
I have tests for +-, whitespace, invalid unit...etc
I'm now using ToIntegral. 
I've added an `assert(default_multiplier > 0)` but I don't know how to test this. Any ideas examples?

Reply:
You cannot have the test trigger an `assert()` and continue successfully. Instead, you can `throw std::out_of_range` from the function and `try/catch` it from the test (`BOOST_CHECK_THROW()`). But in this case maybe it is best/simpler to just remove the parameter. Then there is no need to test edge cases for it :)